src/smt/smt_engine_stats.cpp

```diff
@@ -20,53 +20,29 @@ namespace cvc5 {
 namespace smt {
 
 SmtEngineStatistics::SmtEngineStatistics()
-    : d_definitionExpansionTime("smt::SmtEngine::definitionExpansionTime"),
-      d_numConstantProps("smt::SmtEngine::numConstantProps", 0),
-      d_cnfConversionTime("smt::SmtEngine::cnfConversionTime"),
-      d_numAssertionsPre("smt::SmtEngine::numAssertionsPreITERemoval", 0),
-      d_numAssertionsPost("smt::SmtEngine::numAssertionsPostITERemoval", 0),
-      d_checkModelTime("smt::SmtEngine::checkModelTime"),
-      d_checkUnsatCoreTime("smt::SmtEngine::checkUnsatCoreTime"),
-      d_solveTime("smt::SmtEngine::solveTime"),
-      d_pushPopTime("smt::SmtEngine::pushPopTime"),
-      d_processAssertionsTime("smt::SmtEngine::processAssertionsTime"),
-      d_simplifiedToFalse("smt::SmtEngine::simplifiedToFalse", 0),
-      d_driverFilename("driver::filename", ""),
-      d_driverResult("driver::sat/unsat", ""),
-      d_driverTotalTime("driver::totalTime", 0.0)
+    : d_definitionExpansionTime(smtStatisticsRegistry().registerTimer(
+        "smt::SmtEngine::definitionExpansionTime")),
+      d_numConstantProps(smtStatisticsRegistry().registerInt(
+          "smt::SmtEngine::numConstantProps")),
+      d_cnfConversionTime(smtStatisticsRegistry().registerTimer(
+          "smt::SmtEngine::cnfConversionTime")),
+      d_numAssertionsPre(smtStatisticsRegistry().registerInt(
+          "smt::SmtEngine::numAssertionsPreITERemoval")),
+      d_numAssertionsPost(smtStatisticsRegistry().registerInt(
+          "smt::SmtEngine::numAssertionsPostITERemoval")),
+      d_checkModelTime(smtStatisticsRegistry().registerTimer(
+          "smt::SmtEngine::checkModelTime")),
+      d_checkUnsatCoreTime(smtStatisticsRegistry().registerTimer(
+          "smt::SmtEngine::checkUnsatCoreTime")),
+      d_solveTime(
+          smtStatisticsRegistry().registerTimer("smt::SmtEngine::solveTime")),
+      d_pushPopTime(
+          smtStatisticsRegistry().registerTimer("smt::SmtEngine::pushPopTime")),
+      d_processAssertionsTime(smtStatisticsRegistry().registerTimer(
+          "smt::SmtEngine::processAssertionsTime")),
+      d_simplifiedToFalse(smtStatisticsRegistry().registerInt(
+          "smt::SmtEngine::simplifiedToFalse"))
```

Comment:
Would be a pretty good candidate for the `prefix` approach in the SmtEngineStatistics constructor.

Reply:
I agree.